tests/unit/script/htmlareaelement.rs

```diff
@@ -0,0 +1,288 @@
+            assert_eq! (top_b, vec[3]);
+        },
+        _ => { assert!(false); },
+    }
+}
+fn inspect_polygon (a: Area, vec: Vec<f32>, size: i32) {
+    match a {
+        Area::Polygon { points } => {
+            assert_eq! (size, points.len() as i32);
+            assert! (points.len() % 2 == 0);
+            assert! (points.len() >= 6);
+            let mut index = 0;
+            for x in vec {
+                assert_eq! (x, points[index]);
+                index = index + 1;
+            }
+        },
+
+        _ => { assert!(false); },
+    }
+}
+
+fn inspect_default (a: Area) {
+    match a {
+        Area::Default => { assert!(true); },
+
+        _=> { assert!(false); },
+    }
+}
+/*Pathological tests*/
+
+#[test]
+fn garbage_input ()
+{
+    let area = Area::get_area (";.,()8.2".to_string());
+    inspect_default (area);
+}
+
+
+#[test]
+fn no_case_matching_input ()
+{
+    let area = Area::get_area ("8.2, 10.2".to_string());
+    inspect_default (area);
+}
+
+#[test]
+fn delemiter_input ()
+{
+    let area = Area::get_area (";,  ;,".to_string());
+    inspect_default (area);
+}
+
+/*Area::Circle tests*/
+#[test]
+fn valid_circle_inputs ()
+{
+    let input = "10.2, 3.4, 5.2";
+    let mut vec = Vec::new ();
+
+    vec.push(10.2);
+    vec.push(3.4);
+    vec.push(5.2);
+    let area = Area::get_area (input.to_string());
+
+    inspect_circle(area, vec);
+}
+
+#[test]
+fn valid_negative_circle_inputs ()
+{
+    let input = "-10.2, -3.4, 5.2";
+    let mut vec = Vec::new ();
+
+    vec.push(-10.2);
+    vec.push(-3.4);
+    vec.push(5.2);
+    let area = Area::get_area (input.to_string());
+
+    inspect_circle(area, vec);
+}
+
+/*Area::Rectangle tests*/
+#[test]
+fn rectangle_valid_input ()
+{
+    let input = "5.2, 1.1, 10.2, 3.4";
+
+    let mut vec = Vec::new ();
+
+    vec.push(5.2);
+    vec.push(1.1);
+    vec.push(10.2);
+    vec.push(3.4);
+
+    let area = Area::get_area (input.to_string());
+
+    inspect_rectangle (area, vec);
+}
+
+#[test]
+fn rectangle_valid_negative_input ()
+{
+    let input = "-10.2, -3.4, -5.2, -1.1";
+
+    let mut vec = Vec::new ();
+
+    vec.push(-10.2);
+    vec.push(-3.4);
+    vec.push(-5.2);
+    vec.push(-1.1);
+
+    let area = Area::get_area (input.to_string());
+
+    inspect_rectangle (area, vec);
+}
+
+#[test]
+fn rectangle_invalid_input ()
+{
+    let input = "5.2, 4.3, 10.2, 1.1.2";
+
+    let mut vec = Vec::new ();
+
+    vec.push(5.2);
+    vec.push(0.0);
+    vec.push(10.2);
+    vec.push(4.3);
+
+    let area = Area::get_area (input.to_string());
+
+    inspect_rectangle (area, vec);
+}
+
+#[test]
+fn rectangle_unordered_input ()
+{
+    let input = "5.2, 1.1, 10.2, 4.3";
+
+    let mut vec = Vec::new ();
+
+    vec.push(5.2);
+    vec.push(1.1);
+    vec.push(10.2);
+    vec.push(4.3);
+
+    let area = Area::get_area (input.to_string());
+
+    inspect_rectangle (area, vec);
+}
+
+/*Area::Rectangle tests*/
+#[test]
+fn polygon_six_points_valid_input ()
+{
```

Comment:
nit: Brace uses to be in the same line as the function name except when there's a `where` clause.